src/containers/Bounty/SubmissionsAndCommentsCard.js

```diff
@@ -15,266 +15,309 @@ import { actions as bountyUIActions } from './reducer';
-            acceptFulfillment={() =>
-              initiateLoginProtection(() =>
-                initiateWalkthrough(() =>
-                  acceptFulfillment(bounty.id, fulfillment_id)
-                )
-              )
-            }
-            initiateLoginProtection={initiateLoginProtection}
-            showModal={showModal}
-            setRatingModal={setRatingModal}
-          />
-        </ListGroup.ListItem>
-      );
-    }, list);
-  };
+export const defaultTab = 'comments';
 
-  const renderComments = () => {
-    return map(comment => {
-      const { text, user, created } = comment;
-      const { name, profile_image, public_address } = user;
-
-      return (
-        <ListGroup.ListItem className={styles.commentItem}>
-          <CommentItem
-            name={name}
-            address={public_address}
-            img={profile_image}
-            text={text}
-            created={created}
-          />
-        </ListGroup.ListItem>
-      );
-    }, comments.list);
-  };
+export const mostInterestingTab = fulfillments => {
+  if (!fulfillments) {
+    return defaultTab;
+  }
 
-  let body = null;
-  let bodyClass = '';
+  if (fulfillments.list.length > 0) {
+    return 'submissions';
+  }
 
-  if (currentTab === 'submissions') {
-    body = <ListGroup>{renderFulfillments(fulfillments.list)}</ListGroup>;
+  return defaultTab;
+};
 
-    if (!fulfillments.list.length) {
-      bodyClass = styles.bodyLoading;
-      body = (
-        <div className={styles.zeroState}>
-          <ZeroState
-            title={'There are 0 submissions'}
-            text={'Submissions to this bounty will appear here.'}
-            iconColor="blue"
-            icon={['fal', 'level-up']}
-          />
-        </div>
-      );
+class SubmissionsAndCommentsCardComponent extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      tabLoaded: false
+    };
+  }
+
+  componentDidMount() {
+    const {
+      setActiveTabAction: setActiveTab,
+      fulfillments,
+      comments
+    } = this.props;
+
+    if (!this.state.tabLoaded && !fulfillments.loading && !comments.loading) {
+      this.setState({
+        tabLoaded: true
+      });
+      setActiveTab(mostInterestingTab(fulfillments));
     }
+  }
 
-    if (bounty.private_fulfillments) {
-      if (fulfillments.list.length) {
-        body = (
-          <React.Fragment>
-            <ListGroup>{renderFulfillments(fulfillments.list)}</ListGroup>
-            <Text
-              alignment="align-center"
-              color="defaultGrey"
-              typeScale="Small"
-            >
-              Submissions to this bounty are hidden.
-              {bountyBelongsToLoggedInUser
-                ? ' The submissions are only visible to you and the fulfiller.'
-                : ' Your submissions are only visible to you and the bounty issuer.'}
-            </Text>
-          </React.Fragment>
-        );
-      }
+  render() {
```

Comment:
The diff from here on out looks a bit garbled. I'll do another pass before this merges to make sure nothing else has changed. The diff is confused because it changed from a function to a class.